src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Deposit.tsx

```diff
@@ -46,10 +51,19 @@ export const Deposit: React.FC<DepositProps> = ({
   const history = useHistory()
   const translate = useTranslate()
   const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
-  const { chainId, assetReference } = query
+  const { chainId, assetReference, contractAddress } = query
   const assetNamespace = 'slip44'
+  const assets = useAppSelector(selectAssets)
   const assetId = toAssetId({ chainId, assetNamespace, assetReference })
 
+  const validatorId = toValidatorId({ chainId, account: contractAddress })
+
+  const opportunityMetadataFilter = useMemo(() => ({ validatorId }), [validatorId])
+
+  const opportunityData = useAppSelector(state =>
```

Comment:
```suggestion
  const opportunityMetadata = useAppSelector(state =>
```

Reply:
We could also remove the apy now that we're selecting opportunity (meta)data, if you feel like it @reallybeard